### PR TITLE
Rename XDL to XBF

### LINKED ISSUE #12: Rename to XBF
New name was decided to be eXtensible Binary Format, as it better represents that the project is an on the wire format. Maybe in the future when we write the IDL for the format that can be called XDL, but for now this project only encompasses the on the wire format.

### PR DESCRIPTION
**Pull request by ahuston-0**
Renames XDL to XBF in code and documentation. Currently, all tests are passing and the git history doesn't look like **too** much of a mess currently, but we will absolutely want to merge or rebase this into #17, #18, #22, and any other branch that is currently going in order to make sure they compile as expected. 

Closes #12

### CODE REVIEW

**Comment by ahuston-0** [comment]
Well its a good sign that CI triggered, means I caught that too in the bulk rename, but I need to fix the directory variable real quick I suppose

**Comment by ahuston-0** [comment]
CI looks like it should all pass. Think we're ready for review maybe?

**Comment by ahuston-0** [comment]
Converting this to a draft... Looks like the git history is indeed a mess. Will roll it back to the main branch and try it one at a time (rename directories, change contents, etc.)

**Comment by ahuston-0** [comment]
So far all good... CI is failing but that's expected.

**Comment by ahuston-0** [comment]
@dkrautha for the sake of making this easier to review, requesting that we merge this into the main-xbf-staging branch that I just created with failing CI. Then in a second PR I do the actual changes to the code (so git doesn't think we've moved all files at once). Once that is all solid we merge it into the actual main branch.

**Comment by dkrautha** [comment]
If I'm understanding correctly that makes sense to me, we just ignore CI until we actually go into main, and use this new branch to do atrocious things lol. I'll go ahead and review what's there for now, you mark it for re-review when its out of draft and ready to go.

File: docs/background.md
Comment (dkrautha) [inline docs/background.md:1]: Based on #5 maybe we just nuke this file? I think having a philosophy / background info document is good, but this one is really out of date / not up to date with the current specification. This isn't a big enough deal to stop the merge, and maybe it needs to go in its own branch and pr.

**Review by dkrautha — APPROVED** [review]

**Comment by ahuston-0** [comment]
Oh oops, I thought I marked it ready again. Let's merge this into staging and I'll set up the next two (staging to main and next rename to staging) once I'm home

File: docs/background.md
Comment (ahuston-0) [reply docs/background.md:1]: I'll open another one for #5 once this is done. Do we want to completely eliminate it or summarize it in the README?

Comment (dkrautha) [reply docs/background.md:1]: I think a summary in the readme is good, but then a full philosophy document or something like that in the docs folder.

Comment (ahuston-0) [reply docs/background.md:1]: Okay cool I'll leave this unresolved until I get that into #5 and then come back and resolve this

**Pull request merged**